letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -344,31 +353,51 @@ elif [ -f /etc/redhat-release ]; then
      RPM_DIST_VERSION=0
   fi
 
-  # Starting to Fedora 29, python2 is on a deprecation path. Let's move to python3 then.
-  # RHEL 8 also uses python3 by default.
-  if [ "$RPM_DIST_NAME" = "fedora" -a "$RPM_DIST_VERSION" -ge 29 -o "$PYVER" -eq 26 ]; then
-    RPM_USE_PYTHON_3=1
-  elif [ "$RPM_DIST_NAME" = "rhel" -a "$RPM_DIST_VERSION" -ge 8 ]; then
-    RPM_USE_PYTHON_3=1
-  elif [ "$RPM_DIST_NAME" = "centos" -a "$RPM_DIST_VERSION" -ge 8 ]; then
-    RPM_USE_PYTHON_3=1
-  else
-    RPM_USE_PYTHON_3=0
-  fi
+  # Handle legacy RPM distributions
+  if [ "$PYVER" -eq 26 ]; then
+    # Check if the automated rebootstrap can be achieved on this system.
+    if ! CheckPython36SCLAvailability; then
+      INTERACTIVE_REBOOTSTRAP=1
+      INTERACTIVE_REBOOTSTRAP_MESSAGE="Enable the SCL repository and try running Certbot again."
+    fi
 
-  if [ "$RPM_USE_PYTHON_3" = 1 ]; then
     Bootstrap() {
-      BootstrapMessage "RedHat-based OSes that will use Python3"
-      BootstrapRpmPython3
+      BootstrapMessage "Legacy RedHat-based OSes that will use Python3"
+      BootstrapRpmPython3Legacy
     }
     USE_PYTHON_3=1
-    BOOTSTRAP_VERSION="BootstrapRpmPython3 $BOOTSTRAP_RPM_PYTHON3_VERSION"
+    BOOTSTRAP_VERSION="BootstrapRpmPython3Legacy $BOOTSTRAP_RPM_PYTHON3_LEGACY_VERSION"
+
+    # Try now to enable SCL rh-python36 for systems already bootstrapped
+    # NB: EnablePython36SCL has been defined along with BootstrapRpmPython3Legacy in certbot-auto
+    EnablePython36SCL
   else
-    Bootstrap() {
-      BootstrapMessage "RedHat-based OSes"
-      BootstrapRpmCommon
-    }
-    BOOTSTRAP_VERSION="BootstrapRpmCommon $BOOTSTRAP_RPM_COMMON_VERSION"
+    # Starting to Fedora 29, python2 is on a deprecation path. Let's move to python3 then.
+    # RHEL 8 also uses python3 by default.
+    if [ "$RPM_DIST_NAME" = "fedora" -a "$RPM_DIST_VERSION" -ge 29 -o "$PYVER" -eq 26 ]; then
```

Comment:
In fact, you spotted a underlying issue here. Testing `"$PYVER" -eq 26` has nothing to do here. It has been tested in the other branch, and concerns legacy distributions, not recent ones.

Remove that condition, parenthesis are not necessary anymore on either one or two conditions per if statement.

Reply:
Great! I sort of gave up on understanding all the logic. I'd love to reduce the scope of all the state blowing around in this script, whether by restructuring or by kicking most of it into a different language (as I've tried to do before).

Reply:
Yes, the script goes insane here ... But we cannot refactor in another language sadly. At this point, the system is not bootstrapped for new users, so you cannot rely on Python availability for instance...

Just to give you some insight nonetheless, here is the logic for RPM distributions:
1) Notice that we are on a RedHat-based system because of `/etc/redhat-release` existence.
2) Check if we are on a very old distribution, which is implied by Python 2.6 availability
2.a) Check if we can enable SCL in a consistent way: if yes, prepare Python3 bootstrap for legacy OSes
2.b) If not, prepare to not upgrade the system, and run Certbot without upgrade if non interactive, to avoid breaking cert renewal on these pathologic cases
3) Check we are on very recent systems (Fedora 29, RHEL 8, CentOS 8) that are deprecating Python 2.7: if yes, prepare new Python 3 bootstrap
4) Fallback to install the classic Python 2.7 bootstrap

Reply:
That's helpful; thanks!